garak/translators/remote.py

```diff
@@ -0,0 +1,117 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+
+""" Translator that translates a prompt. """
+
+
+import logging
+
+from garak.exception import BadGeneratorException
+from garak.translators.base import Translator
+
+
+class RivaTranslator(Translator):
+    """Remote translation using NVIDIA Riva translation API
+
+    https://developer.nvidia.com/riva
+    """
+
+    ENV_VAR = "RIVA_API_KEY"
+    DEFAULT_PARAMS = {
+        "uri": "grpc.nvcf.nvidia.com:443",
+        "function_id": "647147c1-9c23-496c-8304-2e29e7574510",
+    }
+
+    # fmt: off
+    # Reference: https://docs.nvidia.com/nim/riva/nmt/latest/support-matrix.html#models
+    bcp47_support = [
+        "zh", "ru", "de", "es", "fr",
+        "da", "el", "fi", "hu", "it",
+        "lt", "lv", "nl", "no", "pl",
+        "pt", "ro", "sk", "sv", "ja",
+        "hi", "ko", "et", "sl", "bg",
+        "uk", "hr", "ar", "vi", "tr",
+        "id", "cs", "en"
+    ]
+    # fmt: on
+
+    def _load_translator(self):
+        if not (
+            self.source_lang in self.bcp47_support
+            and self.target_lang in self.bcp47_support
+        ):
+            raise BadGeneratorException(
+                f"Language pair {self.source_lang}-{self.target_lang} is not supported for this translator service."
+            )
+
+        import riva.client
+
+        if self.nmt_client is None:
+            auth = riva.client.Auth(
+                None,
+                True,
+                self.uri,
+                [
+                    ("function-id", self.function_id),
+                    ("authorization", "Bearer " + self.api_key),
+                ],
+            )
+            self.nmt_client = riva.client.NeuralMachineTranslationClient(auth)
```

Comment:
To enable local Riva usage instead of the public endpoint, expose `use_ssl` on the translator.
```suggestion
        if self.nmt_client is None:
            auth = riva.client.Auth(
                None,
                self.use_ssl,
                self.uri,
                [
                    ("function-id", self.function_id),
                    ("authorization", "Bearer " + self.api_key),
                ],
            )
            self.nmt_client = riva.client.NeuralMachineTranslationClient(auth)
```